### PR TITLE
Delete message and its history entries

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: engagement_database/engagement_database.py [hunk 1]
```diff
@@ -275,6 +275,25 @@ def delete_doc(self, path, transaction=None):
...
+        Deletes a message and its history entries.
+
+        :param message_id: Id of message to delete.
+        :type message_id: str
+        :param transaction: Transaction to run this update in or None.
+                            If None, writes immediately, otherwise adds the updates to a transaction that will need
+                            to be explicitly committed elsewhere.
+        :type transaction: google.cloud.firestore.Transaction | None
+        """
+        message_snapshot = self._message_ref(message_id).get()
```
Comment (as2388) [inline engagement_database/engagement_database.py:289]: I think this needs to be transactional even if this is not taking place in an externally-managed transaction. This is to ensure that in no circumstances can we delete a message without deleting all of its history too. As an example, you can look at `set_message` in this file, whose implementation ensures that we can't update a message without updating its history.

**Review by as2388 — CHANGES_REQUESTED** [review]

File: engagement_database/engagement_database.py (hunk 1 above)
Comment (DanielMwendwa) [reply engagement_database/engagement_database.py]: Updated, thanks.

File: engagement_database/engagement_database.py [hunk 2]
```diff
@@ -275,6 +275,37 @@ def delete_doc(self, path, transaction=None):
...
+        Deletes a message and its history entries.
+
+        :param message_id: Id of message to delete.
+        :type message_id: str
+        :param transaction: Transaction to run this update in or None.
+                            If None, writes immediately, otherwise adds the updates to a transaction that will need
+                            to be explicitly committed elsewhere.
+        :type transaction: google.cloud.firestore.Transaction | None
+        """
+        message_snapshot = self._message_ref(message_id).get(transaction=transaction)
```
Comment (as2388) [inline engagement_database/engagement_database.py:289]:
```suggestion
        message = self.get_message(message_id, transaction)
```

(renaming to message for consistency with history_entries on the next line, which doesn't mention snapshots)

File: engagement_database/engagement_database.py [hunk 3]
```diff
@@ -275,6 +275,37 @@ def delete_doc(self, path, transaction=None):
...
+        
+        if transaction is None:
+            # If no transaction was given, run all the updates in a new batched-write transaction and flag that
+            # this transaction needs to be committed before returning from this function.
+            transaction = self._client.batch()
+            commit_before_returning = True
+        else:
+            commit_before_returning = False
+
+        if message_snapshot.exists:
```
Comment (as2388) [inline engagement_database/engagement_database.py:300]:
```suggestion
        if message is not None
```
for compatibility with the above suggestions

**Review by as2388 — APPROVED** [review]

**Pull request merged**